examples/10_compare_yaw_optimizers.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2022 NREL
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# See https://floris.readthedocs.io for documentation
+
+from time import perf_counter as timerpc
+import numpy as np
+import matplotlib.pyplot as plt
+from floris.tools import FlorisInterface
+from floris.tools.optimization.yaw_optimization.yaw_optimizer_scipy import (
+    YawOptimizationScipy
+)
+from floris.tools.optimization.yaw_optimization.yaw_optimizer_sr import (
+    YawOptimizationSR
+)
+
+
+"""
+This example compares the SciPy-based yaw optimizer with the new Serial-Refine optimizer.
+
+First, we initialize our Floris Interface, and then generate a 3 turbine wind farm. Next, we create two yaw optimization objects, `yaw_opt_sr` and `yaw_opt_scipy` for the Serial-Refine and SciPy methods, respectively. We then perform the optimization using both methods. Finally, we compare the time it took to find the optimal angles and plot the optimal yaw angles and resulting wind farm powers.
+"""
+
+# Load the default example floris object
+fi = FlorisInterface("inputs/gch.yaml") # GCH model matched to the default "legacy_gauss" of V2
+# fi = FlorisInterface("inputs/cc.yaml") # New CumulativeCurl model
+
+# Reinitialize as a 3-turbine farm with range of WDs and 1 WS
+D = 126.0 # Rotor diameter for the NREL 5 MW
+fi.reinitialize(
+    layout=[[0.0, 5 * D, 10 * D], [0.0, 0.0, 0.0]],
+    wind_directions=np.arange(0.0, 360.0, 3.0), 
+    wind_speeds=[8.0],
```

Comment:
@Bartdoekemeijer does the `YawOptimizationScipy` class handle a multiple wind speed scenario? From looking at the code, I can see where it loops over the wind directions individually, but what happens if the user defines multiple wind speeds as well? Seems like a nested loop over `wd_array` and a `ws_array` would be the most obvious solution, but perhaps not the most efficient.

Reply:
Good point, it doesn't support multiple wind speeds currently. I wasn't sure if we were going to change the way we insert wind directions/wind speeds on the `floris` side, so I've left it at that. The base yaw optimizer code currently has [these lines](https://github.com/Bartdoekemeijer/floris/blob/cf3d7c413bd72f3b22b4512b0f2118add1fcc2d9/src/floris/tools/optimization/yaw_optimization/yaw_optimization_base.py#L142) in the initialization step:

```
        # Check floris options
        if self.fi.floris.flow_field.n_wind_speeds > 1:
            raise NotImplementedError(
                "Optimizer currently does not support more than one wind" +
                " speed. Please assign FLORIS a single wind speed."
            )
```

It should be pretty easy to expand over a range of wind speeds, but there are some subtleties depending on whether we assume `n_wind_directions == n_wind_speeds` (timeseries playthrough, kind of style), or a grid style like currently implemented in `floris`. Let me know what you think would be the best way to proceed!

Reply:
Hey @Bartdoekemeijer , getting back to this, I think keeping it in the grid style currently implemented would be best for now. How difficult would that be to support?

Reply:
Should not be too difficult. I'll take a crack at that and let you know if there are any potential barriers!